### PR TITLE
MGMT-17672: [Docs] Adding device and fleet management sections

### PR DESCRIPTION
**Pull request by fzdarsky**

### CODE REVIEW

File: docs/user/managing-devices.md
Comment (avishayt) [inline docs/user/managing-devices.md:235]:
```suggestion
These providers are described in the following.
```

File: docs/user/managing-devices.md
Comment (avishayt) [inline docs/user/managing-devices.md:254]: Let's not use "main".  Remember that we said tracking repos will be post-MVP.

File: docs/user/managing-devices.md
Comment (avishayt) [inline docs/user/managing-devices.md:293]: We should consider making MountPath required because / is read-only.

**Review by avishayt — COMMENTED** [review]
First part of the review...

File: docs/user/managing-devices.md
Comment (avishayt) [inline docs/user/managing-devices.md:285]: > a Kubernetes cluster

Currently the k8s cluster that the service is running on.

File: docs/user/managing-devices.md
Comment (avishayt) [inline docs/user/managing-devices.md:364]: Again, let's avoid "main"

File: docs/user/managing-devices.md
Comment (fzdarsky) [reply docs/user/managing-devices.md]: Fixed.

File: docs/user/managing-devices.md
Comment (fzdarsky) [reply docs/user/managing-devices.md:7]: > > performs service discovery to find its Flight Control Service instance
> > What does this mean?  The agent just looks in a config file.  Is this some future feature?

Arguably, a DNS lookup is already a form of service discovery, but at some point one of the onboarding paths maybe FDO-based.

Comment (fzdarsky) [reply docs/user/managing-devices.md:7]: > > Let's make sure we have JIRAs open for all of the missing details.  I know we have one for wiping keys during decommissioning mentioned above, but I'm not sure about this one.

Ack. I'll create a Jira to collect which other info we want to transmit as part of `device.status.systemInfo`. So far, we have `architecture` and `operatingSystem`, to technically we satisfy the description :).

File: docs/user/managing-fleets.md
Comment (celdrake) [inline docs/user/managing-fleets.md:105]: This syntax does not seem to match the current implementation.

See the examples in https://github.com/flightctl/flightctl/blob/main/internal/tasks/config_helpers_test.go#L75

File: docs/user/managing-devices.md
Comment (hexfusion) [inline docs/user/managing-devices.md:477]: Today implementation does not align with the docs. beforeUpdating is tracked as `OnBeforeUpdate` and is fired before a file is updated by the config controller vs before updating the system. `OnAfterUpdate` refers to immediately after a file is updated.

File: docs/user/managing-devices.md
Comment (hexfusion) [inline docs/user/managing-devices.md:481]: Currently, when an action fails this is reported via status to the service. But a failure here does not fail the update.

File: docs/user/managing-devices.md
Comment (celdrake) [reply docs/user/managing-devices.md:313]: Actually, this is changing soon based on [this PR](https://github.com/flightctl/flightctl/pull/539) and the associated Jira issue. 

We will allow defining multiple files in a single InlineConfig, each having their own permissions, user, etc settings.

File: docs/user/managing-devices.md
Comment (fzdarsky) [reply docs/user/managing-devices.md:7]: > Ack. I'll create a Jira to collect which other info we want to transmit as part of device.status.systemInfo. 
Done. EDM-505.

File: docs/user/managing-devices.md
Comment (fzdarsky) [reply docs/user/managing-devices.md:129]: Implemented in #390.

File: docs/user/managing-devices.md
Comment (fzdarsky) [reply docs/user/managing-devices.md:137]: Yes. I opened [EDM-507](https://issues.redhat.com/browse/EDM-507).

File: docs/user/managing-devices.md
Comment (fzdarsky) [reply docs/user/managing-devices.md:293]: Good catch! In Secrets the MountPath *is* actually already required. Will fix.

I was going for consistency with the Inline and Git config, but for these using (or defaulting to) absolute paths is deliberate.

What is the behaviour when the mountPath contains a read-only location?

File: docs/user/managing-devices.md
Comment (fzdarsky) [reply docs/user/managing-devices.md:285]: Fixed. Also opened [EDM-508](https://issues.redhat.com/browse/EDM-508) to track support for external secrets.

File: docs/user/managing-devices.md
Comment (fzdarsky) [reply docs/user/managing-devices.md:13]: Probably not, no. Any suggestion how to call this? AWS calls it the "device lobby", for example.

File: docs/user/managing-fleets.md
Comment (fzdarsky) [reply docs/user/managing-fleets.md:105]: Good catch, thank you!

File: docs/user/managing-devices.md
Comment (fzdarsky) [reply docs/user/managing-devices.md:313]: I'll start updating based on the PR.

File: docs/user/managing-fleets.md
Comment (fzdarsky) [reply docs/user/managing-fleets.md:105]: You're right, my mistake. Fixed.

File: docs/user/troubleshooting.md
Comment (fzdarsky) [reply docs/user/troubleshooting.md:10]: `jq` isn't installed by default on Linux/Darwin. Would that be an issue?

File: docs/user/managing-devices.md
Comment (fzdarsky) [reply docs/user/managing-devices.md:483]: Both the original PR #424 and the "fixup" PR #449 call the hook `beforeUpdating` and give an example of `run: /usr/local/bin/create-backup.sh` that only makes sense if the hook gets called before updating the system. So I'm not clear what actually is / isn't implemented.

My suggestion would be to remove this whole section until we've clarified the API.

File: docs/user/managing-devices.md
Comment (celdrake) [reply docs/user/managing-devices.md:13]: Maybe something more explicit? In this context, it could just be "You can list all devices currently waiting to be approved / pending approval...".

Otherwise maybe "pending devices lobby"?

File: docs/user/managing-devices.md
Comment (fzdarsky) [reply docs/user/managing-devices.md:5]: I can, yes. Just wondering whether with most editors supporting auto-wrapping, variable window width, and zooming font size, such manual wrapping is worth the downsides (less readable when auto-wrap < manual wrap, larger diffs, ...)?

File: docs/user/managing-devices.md
Comment (fzdarsky) [reply docs/user/managing-devices.md]: Makes sense, thanks. Updated.

File: docs/user/managing-devices.md
Comment (fzdarsky) [reply docs/user/managing-devices.md]: Done.

File: docs/user/managing-devices.md
Comment (fzdarsky) [reply docs/user/managing-devices.md]: Removed.

File: docs/user/managing-fleets.md
Comment (avishayt) [inline docs/user/managing-fleets.md:49]: > which suspends any automation of these fleets until the problem is fixed.

This isn't how it works today.  It's more of a warning.  Device A will be selected by one of the fleets first.  Then, when the other fleet tries to take ownership, we'll set the Conditions.  However, device A will remain in its original fleet and operations will be allowed to continue.  We can discuss changing this behavior, but that's what it is right now.

File: docs/user/managing-fleets.md
Comment (avishayt) [inline docs/user/managing-fleets.md:110]: We also already support parameters in HTTP config

File: docs/user/managing-fleets.md
Comment (fzdarsky) [reply docs/user/managing-fleets.md]: Thanks. I don't want to change behaviour, just misunderstood how it works. Fixed.

File: docs/user/managing-fleets.md
Comment (fzdarsky) [reply docs/user/managing-fleets.md:110]: In both the repo name and URL suffix? Added.

File: docs/user/managing-devices.md
Comment (lkatalin) [inline docs/user/managing-devices.md:7]: Might be good to link `enrollment certificate` to our getting started guide: https://github.com/flightctl/flightctl/blob/main/docs/user/getting-started.md#create-certificate-signing-request-configuration

**Review by lkatalin — APPROVED** [review]
I read through just the enrollment part and it aligns with my conception of what we are doing for enrollment currently. I don't think this doc needs to be updated after #529  merges, as #529 is dealing with only pre-enrollment commands.

File: docs/user/managing-devices.md
Comment (hexfusion) [inline docs/user/managing-devices.md:204]: Do we need to callout this is a required dependency?

File: docs/user/managing-devices.md
Comment (hexfusion) [inline docs/user/managing-devices.md:498]: There is a bug here in the agent as we can not convert d to days. Tracked via https://issues.redhat.com/browse/EDM-518

**Review by hexfusion — APPROVED** [review]
Generally looks good to me some items are future-looking but inline with goals.

File: docs/user/managing-devices.md
Comment (lkatalin) [reply docs/user/managing-devices.md:498]: I commented on EDM-518 but I can submit a PR for that based on the logic in #529 .

**Pull request merged**
